libvast/src/plugin.cpp

```diff
@@ -380,6 +382,55 @@ store_plugin::make_store(system::accountant_actor accountant,
                                                  std::move(path), name());
 }
 
+// -- loader plugin -------------------------------------------------------------
+
+auto stdin_loader_plugin::make_loader(options, const bool_operator*) const
```

Comment:
Let's talk about the second argument: What do we generally need in loaders? Is it a pointer to the operator control plane?

Reply:
What should the loaders generally be able to do aside from loading? Maybe some form of up-/downstream communication. The only kind of upstream communication that the `stdin` loader is doing right now are `chunk_ptr`s to the parser. I can however imagine loaders having to `warn` when a timeout is reached or something strange happens to the carrier that prevents it from sending further data. But then again IDK if that should be the responsibility of the loader plugin or the `from`/`read` umbrella operators.

Reply:
I replaced the `bool_operator`s with our new `operator_control_plane` in https://github.com/tenzir/vast/pull/2935/commits/1c1f320cf0cc59a67a998ad9a2b54a49987c80eb